bin/pa11y

```diff
@@ -81,29 +90,28 @@ function configureProgram(program) {
 			'-e, --phantomjs <path>',
 			'the path to the phantomjs executable'
 		)
+		.option(
+			'-0, --stdin',
+			'read URL from stdin'
+		)
 		.parse(process.argv);
-	program.url = program.args[0];
+	program.url = program.args[0] || null;
 }
 
-function runProgram(program) {
-	if (!program.url || program.args[1]) {
-		program.help();
-	}
-	var options = processOptions(program);
-	options.log.begin(program.url);
+function runTest(test, options, url, callbacks) {
+	processOutput(options.output, url);
+	options.log.begin(url);
 	try {
-		var test = pa11y(options);
-		test.run(program.url, function(error, results) {
+		test.run(url, function(error, results) {
 			if (error) {
 				options.log.error(error.stack);
-				process.exit(1);
+				return callbacks.error(error);
 			}
 			if (reportShouldFail(program.level, results, program.threshold)) {
-				process.once('exit', function() {
-					process.exit(2);
-				});
+				return callbacks.error(error);
```

Comment:
This return prevents the results from ever being output. If we add the following above then it works and the tests pass again:

```
options.log.results(results, url);
```